Sources/HAP/Server/Configuration.swift

```diff
@@ -4,55 +4,68 @@ import func Evergreen.getLogger
 
 fileprivate let logger = getLogger("hap.device")
 
-func generateIdentifier() -> String {
-    do {
-        return Data(bytes: try Random.generate(byteCount: 6))
-            .map { String($0, radix: 16, uppercase: false) }
-            .joined(separator: ":")
-    } catch {
-        fatalError("Could not generate identifier: \(error)")
+extension Device {
+    
+    static internal func generateIdentifier() -> String {
+        do {
+            return Data(bytes: try Random.generate(byteCount: 6))
+                .map { String($0, radix: 16, uppercase: true) }
+                .joined(separator: ":")
+        } catch {
+            fatalError("Could not generate identifier: \(error)")
+        }
+    }
+
+    // Generate a random four character setup key, used in setupURI and setupHash
+    static internal func generateSetupKey() -> String {
+        return String(arc4random_uniform(1679616), radix:36, uppercase: true)
     }
-}
 
-typealias PrivateKey = Data
+    typealias PrivateKey = Data
```

Comment:
I'm not a fan of scoping this inside Device. The PrivateKey might be useful elsewhere, so the definition should probably move outside of this file. I'm fine with leaving it here for now.

Reply:
Ok, I'll revert Configuration back to a standalone class

Reply:
Re-reading your comment, I guess you just mean moving PrivateKey out of Device, will do that